src/lib.rs

```diff
@@ -1,3 +1,162 @@
-#[test]
-fn it_works() {
+#[cfg(test)]
+mod test {
+    use tokens::{ Lexer, LexerOptions };
+
+    #[test]
+    fn lexer_usage() {
+        // build the lexer once for project environment with extensions.
+        let lexer = Lexer::new(LexerOptions::default(), vec![]);
+
+        // use many times.
+        {
+            let source: String = "{{ var }}".into();
+            for token in lexer.tokens(&source) {
+                println!("{:?}", token);
+            }
+        }
+    }
+}
+
+// I don't know where to put this, keeping it in root for now.
+#[derive(Debug, Default, Clone)]
+pub struct Position {
+    pub line: usize,
+    pub column: usize,
+}
+
+// Named this "tokens", in plural, to mean a place where you should expect
+// to find your tokens.
+// Similar convention would work for "nodes", that's where AST lives, and possibly
+// "instructions", if we decide to go that route.
+pub mod tokens {
```

Comment:
Assume we don't want to separate everything into separate `lexer`, `parser`, `node`, `token` (I tried, it is a bit annoying when things are all over the place).

So, a kind of "normalized" and "completely named" structure would have these names (which show the dependencies):

```
token
node
string_to_token_lexer
token_to_node_parser
node_to_whatever_compiler
```

So lexer is fine (can be either `lexer/token` or `token/node`).

But if then we create `lexer/token`, the parser would have to reach into the guts of lexer to get a token :)

Same kind of thing with compiler.

So, it would be nice to treat `lexer` and `parser` as kind of "detail" how those tokens and nodes are made :)